### ISSUE TITLE
LSP: exclude patterns not respected during formatting

### ISSUE THREAD

**Issue by hdemers** [rvben/rumdl#635]
## Bug

The `exclude` patterns defined in `[tool.rumdl]` (e.g. in `pyproject.toml`) are not respected when the LSP server handles formatting requests. Files matching `exclude` patterns are still reformatted on save.

## Root cause

`lint_document` correctly calls `should_exclude_uri` before running diagnostics, but the `formatting` handler in `src/lsp/server.rs` (around line 1006) does not — it proceeds unconditionally to reformat the document.

## Expected behaviour

A file matching an `exclude` pattern should be a no-op for both linting **and** formatting.

## Suggested fix

Add the same `should_exclude_uri` guard to the `formatting` handler:

```rust
async fn formatting(&self, params: DocumentFormattingParams) -> JsonRpcResult<Option<Vec<TextEdit>>> {
    let uri = params.text_document.uri;

    if self.should_exclude_uri(&uri).await {
        return Ok(None);
    }
    // ...
}
```

## Environment

- rumdl version: latest (`rvben/rumdl` main, as of 2026-05-19)
- Editor: Neovim via the built-in LSP client

**Comment by hdemers** [comment]
Apologies, rogue agent created this.
